### PR TITLE
Update attribute.dd

### PR DESCRIPTION
**Pull request by jmh530**
Re-ordering the text to put a greater emphasis on describing abstract member functions before abstract classes. Also, fleshed out some of the details a little bit more.

### CODE REVIEW

File: spec/attribute.dd [hunk 1]
```diff
@@ -695,20 +695,27 @@ $(P
...
+        Only virtual member functions may be declared abstract; non-virtual 
+        member functions and free-standing functions cannot be declared 
+        abstract.
 )
+
 $(P
-        Classes become abstract if they are defined within an
-        abstract attribute, or if any of the virtual member functions
-        within it are declared as abstract.
+        Classes become abstract if any of the virtual member functions
```
Comment (BBasile) [inline spec/attribute.dd]: if any of **its** member function ~~within it~~ **is** declared declared abstract.

File: spec/attribute.dd [hunk 2]
```diff
@@ -695,20 +695,27 @@ $(P
...
+        within it are declared as abstract or if they are defined within an
+        abstract attribute. 
+        Note that an abstract class may also contain non-virtual member functions.
 )
+
 $(P
-        Non-virtual functions cannot be declared as abstract.
+        If a class is abstract, it cannot be instantiated
```
Comment (BBasile) [inline spec/attribute.dd]: I would prefer this shorter, (and more correct) sentence: `As long as a class contains abstract member functions it cannot be instantiated.`

From your sentence it seems like

```D
class Foo
{
    abstract void foo1();
    abstract void foo2();
}

class Bar: Foo
{
    override void foo1(){}
}

Bar bar = new Bar;
```

is possible, which is obviously not true.

**Comment by jmh530** [comment]
@BBasile I believe I have addressed your comments. I changed the wording slightly where I thought appropriate.

**Comment by BBasile** [comment]
Thx, I see nothing else to say about this PR. The description is a bit better and is still compliant with what the compiler does so LGTM.

**Comment by BBasile** [comment]
It might be worth waiting for resolution | official feedback about https://issues.dlang.org/show_bug.cgi?id=2525. There's a problem when an abstract class implements an interface.

**Comment by jmh530** [comment]
@BBasile I'm cool with that.

**Comment by AndrejMitrovic** [comment]
LGTM. I think [Issue 2525](https://issues.dlang.org/show_bug.cgi?id=2525) is fixable sometime in the near future.

**Pull request merged**